tests/ui/precedence.fixed

```diff
@@ -35,3 +40,20 @@ fn main() {
     let b = 3;
     trip!(b * 8);
 }
+
+struct W(u8);
+impl Clone for W {
+    fn clone(&self) -> Self {
+        W(1)
+    }
+}
+
+fn closure_method_call() {
+    let f = move |x: W| { { x }.clone() };
```

Comment:
This part of the lint would conflict with the following lint for which lang FCP has been proposed:

- https://github.com/rust-lang/rust/pull/136906

Probably what I'd suggest instead is annotating the return type to make the outer block braces required:

```suggestion
    let f = move |x: W| -> _ { { x }.clone() };
```

Reply:
As https://github.com/rust-lang/rust/pull/136906#issuecomment-2729179391 mentions `rustfmt` will remove the extra braces from `|x: W| { { x }.clone() }` so we definitely can't suggest that